### PR TITLE
project import condition changes commit

### PR DESCRIPTION
**Pull request by DhaaraniCIT**

### CODE REVIEW

File: src/app/integration/main/main.component.ts [hunk 1]
```diff
@@ -182,20 +190,34 @@ export class MainComponent implements OnInit {
...
+        return (mappingSetting.destination_field !== MappingDestinationField.ACCOUNT && mappingSetting.destination_field !== MappingDestinationField.BANK_ACCOUNT && mappingSetting.destination_field !== MappingDestinationField.CONTACT && mappingSetting.destination_field !== MappingDestinationField.TAX_CODE);
       });
-    }
+      this.fyleFields = responses[0].filter(field => {
+        return !this.mappingSettings.some(mapping => mapping.source_field === field.attribute_type);
+      });
+      this.xeroFields = responses[1].filter((xeroField: ExpenseField) => {
+        return !this.mappingSettings.some(mapping => mapping.destination_field === xeroField.attribute_type);
+      });
+      if ((this.xeroFields.length !== 0 && this.fyleFields.length !== 0) || importedFieldsFromXero.length !== this.mappingSettings.length) {
```
Comment (ashwin1111) [inline src/app/integration/main/main.component.ts:206]: where are we assigning value for importedFieldsFromXero?

File: src/app/shared/components/configuration/import-settings/import-settings.component.ts [hunk 2]
```diff
@@ -20,6 +20,8 @@ import { PreviewDialogComponent } from '../preview-dialog/preview-dialog.compone
 import { WorkspaceService } from 'src/app/core/services/workspace/workspace.service';
 import { TrackingService } from 'src/app/core/services/integration/tracking.service';
 import { ClickEventAdditionalProperty } from 'src/app/core/models/misc/tracking.model';
+import { ThisReceiver } from '@angular/compiler';
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/import-settings/import-settings.component.ts:23]: remove

File: src/app/shared/components/configuration/import-settings/import-settings.component.ts [hunk 3]
```diff
@@ -129,16 +135,26 @@ export class ImportSettingsComponent implements OnInit, OnDestroy {
...
       const projectMapping = expenseFields.filter((expenseField: any) => expenseField.source_field === FyleField.PROJECT);
-      if (projectMapping.length) {
+      if (projectMapping.length || this.isProjectMapped) {
         this.importSettingsForm.controls.importCustomers.setValue(false);
         this.importSettingsForm.controls.importCustomers.disable();
         this.isImportCustomerDisabled = true;
       } else {
         this.importSettingsForm.controls.importCustomers.enable();
         this.isImportCustomerDisabled = false;
+        // This.isProjectMapped = false;
```
Comment (ashwin1111) [inline src/app/shared/components/configuration/import-settings/import-settings.component.ts:147]: remove commented code

File: src/app/integration/main/main.component.ts (hunk 1 above)
Comment (DhaaraniCIT) [reply src/app/integration/main/main.component.ts:206]: line no. 181

Comment (DhaaraniCIT) [reply src/app/integration/main/main.component.ts:206]: this condition is to fine whether there is a custom mapped field or not

**Review by ashwin1111 — APPROVED** [review]

**Pull request merged**